strongbox-web-core/src/main/java/org/carlspring/strongbox/controller/ArtifactIndexesController.java

```diff
@@ -0,0 +1,149 @@
+                                  @PathVariable String storageId,
+                                  @ApiParam(value = "The repositoryId",
+                                            required = true)
+                                  @PathVariable String repositoryId,
+                                  @ApiParam(value = "The artifactPath")
+                                  @PathVariable String path)
+            throws IOException
+    {
+        if (getStorage(storageId) == null)
+        {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                                 .body("The specified storageId does not exist!");
+        }
+        if (getRepository(storageId, repositoryId) == null)
+        {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                                 .body("The specified repositoryId does not exist!");
+        }
+        if (path != null && !new File(getRepository(storageId, repositoryId).getBasedir(), path).exists())
+        {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                                 .body("The specified path does not exist!");
+        }
+        try
+        {
+            artifactIndexesService.rebuildIndexes(storageId, repositoryId, path);
+
+            return ResponseEntity.ok("The Indexes was successfully rebuilt!");
+        }
+        catch (ArtifactStorageException e)
+        {
+            logger.error(e.getMessage(), e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                                 .body(e.getMessage());
+        }
+    }
+
+    @ApiOperation(value = "Used to rebuild the indexes in storage.",
+                  position = 0)
+    @ApiResponses(value = { @ApiResponse(code = 200,
+                                         message = "The indexes was successfully rebuilt!"),
+                            @ApiResponse(code = 500,
+                                         message = "An error occurred."),
+                            @ApiResponse(code = 404,
+                                         message = "The specified storageId does not exist!") })
+    @PreAuthorize("hasAuthority('MANAGEMENT_REBUILD_INDEXES')")
+    @RequestMapping(value = "{storageId}",
+                    method = RequestMethod.POST,
+                    produces = MediaType.TEXT_PLAIN_VALUE)
+    public ResponseEntity rebuild(@ApiParam(value = "The storageId",
+                                            required = true)
+                                  @PathVariable String storageId)
+            throws IOException
+    {
+        if (getStorage(storageId) == null)
+        {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                                 .body("The specified storageId does not exist!");
+        }
+        try
+        {
+            artifactIndexesService.rebuildIndexes(storageId);
+
+            return ResponseEntity.ok("The Indexes was successfully rebuilt!");
```

Comment:
@kate-novik ,

Could you please correct the response string to:

    return ResponseEntity.ok("The indexes were successfully re-built!");

Here and across the other places below...?

Thanks! :)